### PR TITLE
#78 - Create safe component

### LINKED ISSUE #78: Develop create safe modal
image

- Reuse Create card component 
- isActive: boolean - Show gradient border 
- onPress: () => void
- On hover, increase shadow

State must be managed by modal and not the "CreateCard" component

### PR DESCRIPTION
**Pull request by chanchal-sadhukhan**
## Description
Create Safe Component added

## Screenshot
Screenshot from 2022-01-14 00-50-53

### Manual test cases run
_For each manual test case, list the steps to test or reproduce the PR._
- _Description of manual test 1_
- _Description of manual test 2_
- _Description of manual test 3_

## Platform tested
- [x] Large
- [ ] Tablet
- [ ] Mobile

### CODE REVIEW

**Comment by chanchal-sadhukhan** [comment]
I didn't find the icons for Create Safe and Generic Safe

**Comment by Koushith** [comment]
Add this component to the index.ts file.

**Comment by Koushith** [comment]
> I didn't find the icons for Create Safe and Generic Safe

@cksadhukhan  you can export from here- https://www.figma.com/file/0pLW8sjlRJJHiMK5MvWObX/Safient-Designs?node-id=1151%3A625

File: src/components/primitive/create-safe/create-safe.component.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import styled from "styled-components";
+import { BreakPoints } from "utils";
+import { GenericModal } from "../generic-modal/generic-modal.component";
+import { IconSvg } from "../icon-svg/icon-svg.component";
+import { Card } from "../card/card.component";
+import { Text } from "../text/text.component";
+import { CreateSafeModalProps } from "./create-safe.component.props";
+import { Box } from "../box/box.component";
+
+export const SafeCard = styled(Card)`
```
Comment (Koushith) [inline src/components/primitive/create-safe/create-safe.component.tsx:11]: There is a hover effect in our design. also, add border property.

File: src/components/primitive/create-safe/create-safe.component.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      }}
+    >
+      <Box row>
+        <SafeCard hCenter vCenter>
+          <IconSvg name="lock" size="xLarge" />
+          <SafeText variant="content" tx="common.cryptoSafe" />
+        </SafeCard>
+        <SafeCard hCenter vCenter>
+          <IconSvg name="edit" size="xLarge" />
+          <SafeText variant="content" tx="common.genericSafe" />
```
Comment (Koushith) [inline src/components/primitive/create-safe/create-safe.component.tsx]: Text variant will be small and color will be textLight.

File: src/components/primitive/generic-modal/generic-modal.component.tsx [hunk 3]
```diff
@@ -55,9 +61,27 @@ export const GenericModal: React.FunctionComponent<GenericModalComponentProps> =
...
+              <ModalButton
+                label={{ text: "Continue" }}
+                variant="small"
+                onClick={handleClose}
+              />
+            )}
+            {!!onSubmit && (
+              // @ts-ignore
+              <ModalButton {...onSubmit} />
             )}
```
Comment (Koushith) [inline src/components/primitive/generic-modal/generic-modal.component.tsx:84]: There is something wrong with the button, we can ignore this for now. I will open a new issue. will replace this later.

**Review by dsouzaedison — APPROVED** [review]
Merge once the pending comments are resolved.

**Pull request merged**